### PR TITLE
Update Services examples

### PR DESCRIPTION
**Pull request by pablogs9**
Before merge:

- [x] https://github.com/micro-ROS/rclc/pull/44

### CODE REVIEW

File: rclc/addtwoints_client/main.c [hunk 1]
```diff
@@ -8,70 +11,51 @@
...
+	rcl_node_t node = rcl_get_zero_initialized_node();
+	RCCHECK(rclc_node_init_default(&node, "add_twoints_client_rclc", "", &support));
 
-  const char * client_name = "/addtwoints";
-  
-  rcl_client_options_t client_options = rcl_client_get_default_options();
-  rcl_client_t client = rcl_get_zero_initialized_client();
-  const rosidl_service_type_support_t * client_type_support = ROSIDL_GET_SRV_TYPE_SUPPORT(example_interfaces, srv, AddTwoInts);
+  // create client 
+  rcl_client_t client;
```
Comment (JanStaschulat) [inline rclc/addtwoints_client/main.c:35]:
```suggestion
  rcl_client_t client = rcl_get_zero_initialized_client();
```

File: rclc/addtwoints_client/main.c [hunk 2]
```diff
@@ -8,70 +11,51 @@
 #define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc); return 1;}}
 #define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
 
+example_interfaces__srv__AddTwoInts_Request req;
+example_interfaces__srv__AddTwoInts_Response res;
+
+void client_callback(void * msg, rmw_request_id_t * req_id){
```
Comment (JanStaschulat) [inline rclc/addtwoints_client/main.c:17]:
```suggestion
void client_callback(const void * msg, rmw_request_id_t * req_id){
```

File: rclc/addtwoints_client/main.c [hunk 3]
```diff
@@ -8,70 +11,51 @@
...
+  example_interfaces__srv__AddTwoInts_Response * msgin = (example_interfaces__srv__AddTwoInts_Response * ) msg;
+  printf("Received service response %ld + %ld = %ld. Seq %ld\n", req.a, req.b, msgin->sum, req_id->sequence_number);
+}
+
 int main(int argc, const char * const * argv)
 {
```
Comment (JanStaschulat) [inline rclc/addtwoints_client/main.c:23]:
```suggestion
{
    RCLC_UNUSED(argc);
    RCLC_UNUSED(argv);
```

File: rclc/addtwoints_server/main.c [hunk 4]
```diff
@@ -8,57 +11,49 @@
 #define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc); return 1;}}
 #define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
 
+example_interfaces__srv__AddTwoInts_Request req;
+example_interfaces__srv__AddTwoInts_Response res;
+
+typedef void (* rclc_service_callback_t)(const void *, rmw_request_id_t *, void *);
```
Comment (JanStaschulat) [inline rclc/addtwoints_server/main.c:17]: should not be necessary.

File: rclc/addtwoints_server/main.c [hunk 5]
```diff
@@ -8,57 +11,49 @@
...
-            RCSOFTCHECK(rcl_send_response(&serv,&req_id,&res))
-        }
-    }
-  } while ( true );
-
-  RCCHECK(rcl_service_fini(&serv,&node))
-  RCCHECK(rcl_node_fini(&node))
-
-  return 0;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
```
Comment (JanStaschulat) [inline rclc/addtwoints_server/main.c:31]:
```suggestion
  RCLC_UNUSED(argc);
  RCLC_UNUSED(argv);
  rcl_allocator_t allocator = rcl_get_default_allocator();
```

File: rclc/addtwoints_server/main.c [hunk 6]
```diff
@@ -8,57 +11,49 @@
...
+
+	// create init_options
+	RCCHECK(rclc_support_init(&support, 0, NULL, &allocator));
+
+	// create node
+	rcl_node_t node = rcl_get_zero_initialized_node();
+	RCCHECK(rclc_node_init_default(&node, "add_twoints_client_rclc", "", &support));
+
+  // create service
+  rcl_service_t service;
```
Comment (JanStaschulat) [inline rclc/addtwoints_server/main.c:42]:
```suggestion
  rcl_service_t service = rcl_get_zero_initialized_service();
```

**Review by JanStaschulat — CHANGES_REQUESTED** [review]
Initialization of service/client
const void pointer - first argument of client_callback
compiler warnings -  unused arguments

**Comment by pablogs9** [comment]
All addressed

**Review by JanStaschulat — APPROVED** [review]

**Pull request merged**